### DISCUSSION TITLE
Automatically Closing Local Session During Remote Connection on Plasma

### DISCUSSION THREAD

**Discussion by steven12138** [neutrinolabs/xrdp#3269] [Q&A]
When remotely connecting using Plasma 6.1.5 on Arch Linux with xrdp version 0.10.0, if the local session is still active (not logged out), the remote connection displays a black screen with a cursor, while some newly opened windows appear in the remote session. I want to automatically close the local session (X11 or Wayland) when initiating a remote connection. Is this possible, and if so, how can it be achieved?

I tried
```bash
dbus-launch --exit-with-session
startplasma-x11
```
but its not what I'm expected

I tried follow:
https://github.com/neutrinolabs/xrdp/discussions/3001
and remove the '.' character in 
![image](https://github.com/user-attachments/assets/086ca581-6bba-47e0-b928-8afba82ab595)
and here is my `~/.xinitrc`
```bash
dbus-launch --exit-with-session
startplasma-x11
```

**Comment by matt335672** [comment]
This is a problem on systemd-based systems. See [the FAQ](https://github.com/neutrinolabs/xrdp/wiki/Tips-and-FAQ#why-cant-i-log-the-same-user-on-on-the-graphical-console-and-over-xrdp-at-the-same-time).

If at all possible, use a different user for xrdp and the console. That's the official advice for such systems.

If not, this might work for you. I've not tested it on arch.

Create a script `/etc/xrdp/kill_seats.sh` with the following contents:-

```sh
#!/bin/sh

# Check for a systemd installation
if ! [ -x /usr/bin/loginctl ]; then
    exit 0
fi

# Need zenity to prompt the user
if ! command -v zenity >/dev/null; then
    echo "** zenity not installed" >&2
    exit 0
fi

status=0
seat_list=""
session_list=""
for session in $(loginctl show-user "$(id -un)"  -p Sessions --value); do
    seat="$(loginctl show-session "$session" -p Seat --value)"
    if [ -n "$seat" ]; then
        seat_list="$seat_list $seat"
        session_list="$session_list $session"
    fi
done

if [ -n "$seat_list" ]; then
    seat_list="${seat_list# }" ;# Remove leading ' '
    zenity --display="$DISPLAY" --question \
        --text='Seat(s) "'"$seat_list"'" are active for this user\nDo you want to kill them?\nIf you do not, this xrdp session will terminate.'
    status=$?
    if [ $status -eq 0 ]; then
        # 'Yes' pressed
        # shellcheck disable=SC2086
        loginctl terminate-session $session_list
        status=$?
    fi
fi

exit $status
```

Make sure the script is owned by root, and executable.

Within `/etc/xrdp/startm.sh`, near the top, add these lines:-

```sh
# Check the console isn't logged in
/etc/xrdp/kill_seats.sh || exit $?
```

Then install zenity.

On my test machine running 22.04, I needed to use CTRL-Alt-F1 on the console to get the login box back after terminating a console session. I have no idea why.
